lib/core/scheduler.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "logger.hpp"
+#include "time.hpp"
+#include "types.hpp"
+
+#include <cstdint>
+#include <functional>
+#include <queue>
+#include <unordered_map>
+
+namespace hyped::core {
+
+class Scheduler {
+ public:
+  Scheduler(core::ILogger &logger, core::ITimeSource &time);
+  /**
+   * @brief Calls the task at the top of the queue if current time is greater than the delay
+   * specified by the task
+   *
+   * @return kSuccess if no task is called, otherwise returns return value of the task
+   */
+  core::Result run();
```

Comment:
I think this should run *all* the tasks that are due.

Further, the return value does not differentiate whether (a) no task was run or (b) a task was run successfully. I believe that this is an important distinction.